### PR TITLE
BAQE-1112 - Create new runtime benchmark using decision tables

### PR DESCRIPTION
**Pull request by hellowdan**
BAQE-1112 - Create new runtime benchmark using decision tables

### CODE REVIEW

File: drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private static int to = 3102;
+
+    @Setup
+    public void addResources() {
+        addClassPathResource("kbase-creation/dtable1-kbase-creation.xls");
+    }
+
+    @Setup(Level.Iteration)
+    public void createKieSession() {
+        ksession = RuntimeUtil.createKieSession(kieBase, ClockTypeOption.get("pseudo"));
```
Comment (baldimir) [inline drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java]: I know we talked about extracting session creation outside the benchmark, but please remove this createKieSession() method and insertFactsAndFireAllRules() method, so the ones from the abstract class are used. If the session creation is part of the benchmark method in the abstract class (therefore in other benchmarks), we should use it the same way. What do you think?

File: drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.conf.ClockTypeOption;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Setup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class KBaseCreationFromDTable1Benchmark extends AbstractSimpleRuntimeBenchmark {
```
Comment (baldimir) [inline drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java]: Please tweak the warmup and measurement iterations here - the ones in the parent are just the default ones. Usually each benchmark needs its specific number of warmup and measurement iterations.

File: drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable2Benchmark.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+public class KBaseCreationFromDTable2Benchmark extends AbstractSimpleRuntimeBenchmark {
+
+    @Setup
+    public void addResources() {
+        addClassPathResource("kbase-creation/dtable2-kbase-creation.xls");
+    }
+
+    @Setup(Level.Iteration)
+    public void createKieSession() {
+        ksession = RuntimeUtil.createKieSession(kieBase, ClockTypeOption.get("pseudo"));
```
Comment (baldimir) [inline drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable2Benchmark.java]: Same comment as in the previous benchmark class. Please remove this and insertFactsAndFireAllRules() method.

File: drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable2Benchmark.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import org.drools.benchmarks.common.util.RuntimeUtil;
+import org.drools.benchmarks.turtle.runtime.generator.GeneratorConfiguration;
+import org.drools.benchmarks.turtle.runtime.generator.KBaseCreationFromDTable2Generator;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.conf.ClockTypeOption;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Setup;
+
+public class KBaseCreationFromDTable2Benchmark extends AbstractSimpleRuntimeBenchmark {
```
Comment (baldimir) [inline drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable2Benchmark.java]: Please tweak the warmup and measurement iterations here - the ones in the parent are just the default ones. Usually each benchmark needs its specific number of warmup and measurement iterations.

File: drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java (hunk 1 above)
Comment (hellowdan) [reply drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java]: I agree. The main reason I override it was to catch how many rules were fired. But there's nothing more being done so there's no need to override it.

File: drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java (hunk 2 above)
Comment (hellowdan) [reply drools-benchmarks/src/main/java/org/drools/benchmarks/turtle/runtime/KBaseCreationFromDTable1Benchmark.java]: @Warmup(iterations = 60, time = 50)
Warmup Iteration 1: 41.254 ms/op
After 40 iterations it’s stable on ~5 ms/op

@Measurement(iterations = 20, time = 10)
No more time than 10 ms (sometimes it reaches 7.539 ms/op)

**Pull request merged**